Tasks/SqlAzureDacpacDeploymentV1/Utility.ps1

```diff
@@ -190,22 +181,43 @@ function Get-SqlPackageCommandArguments
         $sqlServerName = $targetServerName
 
         if (-not $databaseName) {
-          $databaseName = $sourceDatabaseName
+            $databaseName = $sourceDatabaseName
         }
 
         if (-not $sqlServerName) {
-          $sqlServerName = $sourceServerName
+            $sqlServerName = $sourceServerName
         }
 
         $connectionString = Get-AADAuthenticationConnectionString -authenticationType $authenticationType -serverName $sqlServerName -databaseName $databaseName -sqlUserName $sqlUserName -sqlPassword $sqlPassword
 
         if ($targetDatabaseName) {
-          $sqlPackageArguments += @("$($sqlPackageOptions.TargetConnectionString)`"$connectionString`"")
+            $sqlPackageArguments += @("$($sqlPackageOptions.TargetConnectionString)`"$connectionString`"")
         }
         else {
-          $sqlPackageArguments += @("$($sqlPackageOptions.SourceConnectionString)`"$connectionString`"")
+            $sqlPackageArguments += @("$($sqlPackageOptions.SourceConnectionString)`"$connectionString`"")
         }
     }
+    elseif ($authenticationType -eq "servicePrincipal") {
+        if (!($SourceServerName -or $targetServerName)) {
+            throw "No server specified"
```

Comment:
add error message in resources.resjson. Use like this `throw  (Get-VstsLocString`

Reply:
Added ```SAD_NoServerSpecified```